app/src/main/java/com/space/quiz_app/presentation/quiz_gpa_screen/ui/QuizGPAFragment.kt

```diff
@@ -18,14 +19,21 @@ class QuizGPAFragment : QuizBaseFragment<QuizGPAViewModel>() {
     override val viewModelClass: KClass<QuizGPAViewModel>
         get() = QuizGPAViewModel::class
 
+    override fun onCreateFragment() {
+
+    }
+
     override fun onBind() {
         navigate()
     }
 
     private fun navigate() {
-        // This is just to navigate onto the next screen and test it on the actual device
+        // TODO navigation will be written in ViewModel
         binding.backButton.setOnClickListener {
-            findNavController().navigate(QuizGPAFragmentDirections.actionTakenQuizFragmentToHomeFragment())
+            findNavController().navigate(QuizGPAFragmentDirections.actionGlobalHomeFragment())
+        }
+        requireActivity().onBackPressedDispatcher.addCallback {
+            findNavController().navigate(QuizGPAFragmentDirections.actionGlobalHomeFragment())
```

Comment:
what is requireActivity().onBackPressedDispatcher.addCallback and why do you need

Reply:
To navigate back from GPA to Home fragment when I press android back button.

Reply:
I don't think so. When I'll get there, I'll know better

Reply:
<img width="441" alt="Screenshot 2023-06-15 at 19 26 25" src="https://github.com/DavitTsiskarashvili/Space-Quiz-App/assets/70803715/4c56474a-c650-41d0-a23f-28cedd018aec">
you need callback here

Reply:
You're right. I need that callback to close that dialog and also during quiz if user presses back button an exit dialog should appear

Reply:
then fix your callback issue

Reply:
Giorgi I don't get you. What is the issue? I'm still not at that stage of the project